frappe/printing/doctype/letter_head/letter_head.py

```diff
@@ -8,21 +8,19 @@
 
 
 class LetterHead(Document):
-	def before_insert(self):
-		# for better UX, let user set from attachment
-		self.source = "Image"
-
 	def validate(self):
 		self.set_image()
-		self.validate_disabled_and_default()
-
-	def validate_disabled_and_default(self):
+		self.use_as_default()
 		if self.disabled and self.is_default:
 			frappe.throw(_("Letter Head cannot be both disabled and default"))
 
-		if not self.is_default and not self.disabled:
-			if not frappe.db.exists("Letter Head", dict(is_default=1)):
-				self.is_default = 1
+	def use_as_default(self):
+		if (
+			not self.is_default
+			and not self.disabled
+			and not frappe.db.exists("Letter Head", dict(is_default=1))
+		):
+			self.is_default = 1
```

Comment:
this will let us uncheck default checkbox on a default letterhead right? ideally that should not happen right?

Reply:
> Not sure if I get your question

so that logic will surely set the current letter to default but it will also let us uncheck the same as well

this becomes problematic:
a) if there is only 1 letterhead, things will very likely be fine
b) if there are multiple letterheads, the previous default letter head (if present) is un-defaulted implicitly

imo user should choose another letter head they want to make default upon which a confirm dialog box should show up whether they want to make this the default letter head as another is already present.

This adds a bit of curve to the current process but something i'd very much like to have.

#

> On the other hand, I'm not sure why we need this logic at all. Whats so bad about not having a default? IMO we could probably just as well remove this.

this is a valid point. but we use `frappe.defaults.get_default("letter_head")` in various print formats in erpnext hence we're stuck with it for the moment i guess.

which also brings me to a question: was a default letter head always supposed to exist? since erpnext probably assumes that

Reply:
Also there's some duplication, since erpnext's **Company** has _Default Letter Head_ as well and it's unclear to the user what is used when. Thinking of replacing occurrences of `frappe.defaults.get_default("letter_head")` with more nuanced logic.

Reply:
@phot0n this is still unresolved. IMO, the default letter head should be determined via the Company, in the context of ERPNext. What do you think?

Reply:
would having this as fallback make sense in that case? if not found in company, resort to the global default letterhead?

Reply:
> But can we still allow not having a global default?

so i had a chat with @surajshetty3416 regarding this - the current behaviour of having a default checkbox and not enforcing the default is fine at the framework level. 

if erpnext needs certain specific thing, it needs to enforce it.